tests/storage/sqlite_dos/test_backend.py

```diff
@@ -25,3 +25,15 @@ def test_archive_import(aiida_config, aiida_profile_factory):
         assert QueryBuilder().append(Node).count() == 0
         import_archive(get_archive_file('calcjob/arithmetic.add.aiida'))
         assert QueryBuilder().append(Node).count() > 0
+
+
+def test_backup(aiida_config, aiida_profile_factory, tmp_path, manager):
+    """Test the backup implementation."""
+    with aiida_profile_factory(aiida_config, storage_backend='core.sqlite_dos'):
+        storage = manager.get_profile_storage()
+        storage.backup(str(tmp_path))
```

Comment:
No strictly related to this PR but why doesn't storage backup accept Path?

Reply:
This is because Gio wanted to have support for paths over ssh, so it should support strings of the form `user@some-host:/some/path`. I guess the implementation could also accept `pathlib.Path` values and then convert to string as needed. But the implementation is in `disk-objectstore` and so that would have to be updated first.

Reply:
Right, to clarify I wanted to suggest that the menthod should accept both so that the logic is not pushed to the callers.

Reply:
https://github.com/aiidateam/disk-objectstore/issues/170